### PR TITLE
docs: Add filesystem POSIX requirement

### PR DESCRIPTION
**Pull request by brancz**
Officially adding this to our docs (we've already [commented this on issues](https://github.com/prometheus/prometheus/issues/3534#issuecomment-348598966), but I wanted to call it out explicitly in our docs).

@brian-brazil 

Signed-off-by: Frederic Branczyk <fbranczyk@gmail.com>

### CODE REVIEW

File: docs/storage.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ needed_disk_space = retention_time_seconds * ingested_samples_per_second * bytes
...
 
-If your local storage becomes corrupted for whatever reason, your best bet is to shut down Prometheus and remove the entire storage directory. However, you can also try removing individual block directories to resolve the problem. This means losing a time window of around two hours worth of data per block directory. Again, Prometheus's local storage is not meant as durable long-term storage.
+If your local storage becomes corrupted for whatever reason, your best bet is to shut down Prometheus and remove the entire storage directory. Non POSIX compliant filesystems are not supported by Prometheus's local storage, corruptions may happen, without possibility to recover. However, you can also try removing individual block directories to resolve the problem. This means losing a time window of around two hours worth of data per block directory. Again, Prometheus's local storage is not meant as durable long-term storage.
```
Comment (brian-brazil) [inline docs/storage.md]: We should explicitly call out NFS here.

Comment (brancz) [reply docs/storage.md]: As in "Non POSIX compliant filesystems (such as NFS) ..." ?

Comment (brian-brazil) [reply docs/storage.md]: NFS is (potentially) POSIX, most implementations are not.

Comment (brancz) [reply docs/storage.md]: Added

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**